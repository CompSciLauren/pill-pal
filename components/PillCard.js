import React from 'react';
import { Platform, StyleSheet, Text, View } from 'react-native';

export function PillCard(props) {
  const { name, formattedTimeLeft, dosage } = props;
  return (
    <View style={styles.pillCardInfoContainer}>
      <View style={{ flex: 3 }}>
        <Text style={styles.pillCardTitleText}>
          {name}: {formattedTimeLeft}
        </Text>

        <Text style={styles.pillCardDosageText}>Dosage: {dosage}</Text>
      </View>

      <View style={{ flex: 1, flexDirection: 'row' }}>
        <Text style={styles.pillCardLogPillText}>Log Pill</Text>

        <Text style={styles.pillCardDismissText}>Dismiss</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  pillCardInfoContainer: {
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 10,
    marginRight: 10,
    height: 125,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { width: 0, height: -3 },
        shadowOpacity: 0.1,
        shadowRadius: 3,
      },
      android: {
        elevation: 5,
      },
    }),
    alignItems: 'flex-start',
    backgroundColor: '#fbfbfb',
    borderRadius: 10,
    paddingVertical: 15,
    paddingHorizontal: 15,
  },
  pillCardTitleText: {
    fontSize: 17,
    color: 'rgba(70,70,70, 1)',
    textAlign: 'left',
  },
  pillCardDosageText: {
    fontSize: 13,
    color: 'rgba(96,100,109, 0.8)',
    textAlign: 'left',
  },
  pillCardLogPillText: {
    fontSize: 18,
    fontWeight: 'bold',
    color: 'rgba(121,51,153, 1)',
    textAlign: 'left',
  },
  pillCardDismissText: {
    marginLeft: 60,
    fontSize: 18,
    fontWeight: 'bold',
    color: 'rgba(121,51,153, 1)',
    textAlign: 'left',
  },
});
