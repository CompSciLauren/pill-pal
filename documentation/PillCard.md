## PillCard

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**dosage** | `dosage` |  | :x: | The quantity of pills to be taken.
**formattedTimeLeft** | `formattedTimeLeft` |  | :x: | The amount of time left before the user needs to take their next medication.
**name** | `name` |  | :x: | The name of the pill to take.

