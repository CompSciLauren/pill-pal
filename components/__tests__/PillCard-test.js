import React from 'react';
import renderer from 'react-test-renderer';

import { PillCard } from '../PillCard';

it(`renders correctly`, () => {
  const tree = renderer
    .create(<PillCard name="name" formattedTimeLeft="time" dosage="dosage" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
