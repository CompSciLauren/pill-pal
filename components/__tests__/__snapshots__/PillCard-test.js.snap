// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fbfbfb",
      "borderRadius": 10,
      "height": 125,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingHorizontal": 15,
      "paddingVertical": 15,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": -3,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(70,70,70, 1)",
          "fontSize": 17,
          "textAlign": "left",
        }
      }
    >
      name
      : 
      time
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(96,100,109, 0.8)",
          "fontSize": 13,
          "textAlign": "left",
        }
      }
    >
      Dosage: 
      dosage
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(121,51,153, 1)",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
    >
      Log Pill
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(121,51,153, 1)",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginLeft": 60,
          "textAlign": "left",
        }
      }
    >
      Dismiss
    </Text>
  </View>
</View>
`;
